tests/python/gaia-ui-tests/gaiatest/apps/music/regions/player_view.py

```diff
@@ -0,0 +1,102 @@
+
+        #wait until the overlay disappears
+        Wait(self.marionette).until(expected.element_not_displayed(*self._rating_view_locator))
+        self.marionette.switch_to_shadow_root()
+
+        self.marionette.find_element(*self._cover_image_shadow_DOM_locator).tap()
+        self.marionette.switch_to_shadow_root(self.marionette.find_element(*self._cover_image_shadow_DOM_locator))
+        Wait(self.marionette).until(expected.element_displayed(
+            Wait(self.marionette).until(expected.element_present(*self._rating_view_locator))))
+        self.apps.switch_to_displayed_app()
+
+    def tap_share_button(self):
+        self.marionette.switch_to_frame(self.marionette.find_element(*self._active_view_locator))
+        self.marionette.switch_to_shadow_root(self.marionette.find_element(*self._cover_image_shadow_DOM_locator))
+        self.marionette.find_element(*self._cover_share_locator).tap()
+        from gaiatest.apps.system.regions.activities import Activities
+        return Activities(self.marionette)
+
+    def _get_star_locator(self, rating):
+        return By.CSS_SELECTOR, 'button[value="%s"]' % rating
+
+    def tap_star(self, rate):
+        """
+        give rating.  (After tapping the cover to make the ratings overlay appear)
+        """
+        self.tap_cover_in_player_view()
+        self.marionette.switch_to_frame(self.marionette.find_element(*self._active_view_locator))
+        self.marionette.switch_to_shadow_root(self.marionette.find_element(*self._cover_image_shadow_DOM_locator))
+        self.marionette.switch_to_shadow_root(self.marionette.find_element(*self._rating_view_locator))
+        Wait(self.marionette).until(expected.element_displayed(*self._get_star_locator(rate)))
+        self.marionette.find_element(*self._get_star_locator(rate)).tap()
+        self.apps.switch_to_displayed_app()
+
+    @property
+    def star_rating(self):
+        self.marionette.switch_to_frame(self.marionette.find_element(*self._active_view_locator))
+        self.marionette.switch_to_shadow_root(self.marionette.find_element(*self._cover_image_shadow_DOM_locator))
+        rating = self.marionette.find_element(*self._rating_view_locator).get_attribute('value')
+        self.apps.switch_to_displayed_app()
+        return int(rating)
+
+    @property
+    def player_elapsed_time(self):
+        self.marionette.switch_to_frame(self.marionette.find_element(*self._active_view_locator))
+        self.marionette.switch_to_shadow_root(self.marionette.find_element(*self._seek_bar_shadow_DOM_locator))
+        Wait(self.marionette).until(expected.element_displayed(
+            Wait(self.marionette).until(expected.element_present(*self._player_seek_elapsed_locator))))
+        elapsed_time = time.strptime(self.marionette.find_element(*self._player_seek_elapsed_locator).text, '%M:%S')
+        self.apps.switch_to_displayed_app()
+        return elapsed_time
+
+    def is_player_playing(self):
```

Comment:
Oh I remember this function having intermittency. Could we change it to something like:
``` py
def wait_until_time_stamp_has_changed(self):
    base_timestamp = self.player_current_timestamp
    Wait(self.marionette).until(lambda m: self.player_current_timestamp > base_timestamp)
```
Hence we can solve [bug 1201880](https://bugzilla.mozilla.org/show_bug.cgi?id=1201880), directly?